src/number/sum.ts

```diff
@@ -14,10 +14,13 @@
  *   {value: 3}
  * ], (item) => item.value)
  * // => 6
+ * 
+ * sum([true, false, true], (item) => item ? 1 : 0)
+ * // => 2
  * ```
  */
 export function sum<T extends number>(array: readonly T[]): number
-export function sum<T extends object>(
+export function sum<T extends object | boolean>(
   array: readonly T[],
   fn: (item: T) => number,
 ): number
```

Comment:
The old `sum` types were bad, so I'd suggest a more ambitious change:
```ts
export function sum(array: readonly number[]): number
export function sum<T>(
  array: readonly T[],
  fn: (item: T) => number,
): number
```

Reply:
I also prefer your suggestion. It doesn't limit the typing.

Reply:
Unfortunately without removing any I couldn't get typing to create a valid overload through TypeScript. But I think it improved a bit.